redpen-core/src/main/java/cc/redpen/parser/PreprocessorRule.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * redpen: a text inspection tool
+ * Copyright (c) 2014-2015 Recruit Technologies Co., Ltd. and contributors
+ * (see CONTRIBUTORS.md)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cc.redpen.parser;
+
+import cc.redpen.model.Document;
+import cc.redpen.model.ListBlock;
+import cc.redpen.model.ListElement;
+import cc.redpen.model.Paragraph;
+import cc.redpen.model.Section;
+import cc.redpen.model.Sentence;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * An preprocessor 'rule'. Supported rule types are:
+ *
+ * SUPPRESS: Suppress the output of validation errors for a section, block or paragraph
+ */
+public class PreprocessorRule {
+
+    public enum RuleType {
+        SUPPRESS
+    }
+
+    private int lineNumber = 0;
+    private int lineNumberLimit = 0;
+    private List<String> parameters = new ArrayList<>();
+    private RuleType ruleType;
+
+    public PreprocessorRule(RuleType ruleType, int lineNumber) {
+        this.ruleType = ruleType;
+        this.lineNumber = lineNumber + 1; // applies the to line after the line the rule was detected
+    }
+
+    public RuleType getRuleType() {
+        return ruleType;
+    }
+
+    public int getLineNumber() {
+        return lineNumber;
+    }
+
+    public int getLineNumberLimit() {
+        return lineNumberLimit;
+    }
+
+    public void setLineNumberLimit(int lineNumberLimit) {
+        this.lineNumberLimit = lineNumberLimit;
+    }
+
+    public List<String> getParameters() {
+        return parameters;
+    }
+
+    public void addParameter(String parameter) {
+        parameters.add(parameter.toLowerCase());
+    }
+
+    /**
+     * Return true if the rule is triggered by the given line and name given the structure
+     * of the supplied document
+     *
+     * @param document
+     * @param otherLineNumber
+     * @param name
+     * @return
+     */
+    public boolean isTriggeredBy(Document document, int otherLineNumber, String name) {
```

Comment:
I would like to have test cases including corner cases, to understand the expected behavior.